READMEにテーブル情報を記載しました。

yasukuro6259 opened the pull request:
#WHAT
テーブル情報を記載しました。

#WHY
データベースを構築するため

yasukuro6259 committed:
READMEにテーブル情報を記載しました。

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+| family_name    | string  | null: false |
+| ja_first_name  | string  | null: false |
+| ja_family_name | string  | null: false |
+| birth_year     | integer | null: false |
+| birth_month    | integer | null: false |
+| birth_day      | integer | null: false |
```
| birth_year     | integer | null: false |
| birth_month    | integer | null: false |
| birth_day      | integer | null: false |

date型にし、一つのカラムにまとめましょう！

yasukuro6259 replied:
修正しました！

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+
+| Column         | Type      | Options                        |
+| -------------- | --------- | ------------------------------ |
+| image          | string    | null: false                    |
```
| image          | string    | null: false                    |
active_storageを用いるので、このカラムは不要です！

yasukuro6259 replied:
修正しました！

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+| name           | string    | null: false                    |
+| description    | string    | null: false                    |
+| price          | integer   | null: false                    |
+| category       | string    | null: false                    |
+| condition      | string    | null: false                    |
+| fee            | integer   | null: false                    |
+| region         | string    | null: false                    |
+| shipping_days  | integer   | null: false                    |
```
| category       | string    | null: false                    |
| condition      | string    | null: false                    |
| fee            | integer   | null: false                    |
| region         | string    | null: false                    |
| shipping_days  | integer   | null: false                    |

active_hashを用いるので、_idをつけ、integer型にしましょう！

yasukuro6259 replied:
修正しました！

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+
+| Column        | Type       | Options                        |
+| ------------- | ---------- | ------------------------------ |
+| credit_number | integer    | null: false                    |
+| cvc           | integer    | null: false                    |
+| exp_month     | integer    | null: false                    |
+| exp_year      | integer    | null: false                    |
```
| credit_number | integer    | null: false                    |
| cvc           | integer    | null: false                    |
| exp_month     | integer    | null: false                    |
| exp_year      | integer    | null: false                    |

セキュリティ的観点から、カード情報は通常DBに保存しません。
そのため、上記カラムは削除しましょう！

yasukuro6259 replied:
修正しました！

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+| Column        | Type       | Options                        |
+| -------       | ---------- | ------------------------------ |
+| postal_code   | string     | null: false                    |
+| prefecture    | string     | null: false                    |
```
| prefecture    | string     | null: false                    |

active_hashを用いるので、integer型にしましょう！

isyt729 commented on README.md:
```diff
@@ -22,3 +22,94 @@ Things you may want to cover:
+| prefecture    | string     | null: false                    |
+| city          | string     | null: false                    |
+| address_line  | string     | null: false                    |
+| phone_number  | integer    | null: false                    |
```
| phone_number  | integer    | null: false                    |

integer型だと０から始まる電話番号を正しく保存できないため、stringにしましょう！

yasukuro6259 replied:
修正しました！

isyt729 reviewed (commented):
レビュー依頼ありがとうございます！
コメントしましたので、ご確認お願いします！

また、プルリクエストないにER図がありませんので、貼り付けていただくようにお願いします！

yasukuro6259 commented:
#ER図のURL
ER図を作成しましたので、下記URLからご確認の程、どうぞよろしくお願いいたします。
https://app.lucidchart.com/invitations/accept/2d4f9e90-a6ed-4ab1-8c9d-c185aeb9fdb0
また、おすすめのフリーのER図作成ツールがあればご教示いただけないでしょうか。

yasukuro6259 committed:
README修正しました。

usersテーブル 生年月日部分のカラムをdate型に変更し、一つにまとめた
itemsテーブル imageカラムの削除、active_hashを用いるカラム名の修正
purchase_itemsテーブル カード情報カラムの削除
addressesテーブル active_hashを用いるカラム名の修正、電話番号カラムをstring型に修正

Katsuno0814 commented on README.md:
```diff
@@ -48,18 +46,17 @@ Things you may want to cover:
+| Column           | Type      | Options                        |
+| ---------------- | --------- | ------------------------------ |
+| name             | string    | null: false                    |
+| description      | string    | null: false                    |
```
商品説明を保存するカラムは、説明が長くなることを仮定すると、text型で設計した方が無難かと考えます。

yasukuro6259 replied:
修正しました！

Katsuno0814 commented on README.md:
```diff
@@ -104,10 +97,10 @@ Things you may want to cover:
-| prefecture    | string     | null: false                    |
+| prefecture_id | string     | null: false                    |
 | city          | string     | null: false                    |
 | address_line  | string     | null: false                    |
```
こちら建物番号を保存するカラムを用意いたしましょう！

yasukuro6259 replied:
修正しました！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

yasukuro6259 committed:
README修正しました。（2回目）

商品説明カラムをtext型に変更
addressesテーブルにbuilding_numberカラムを追記

Katsuno0814 commented on README.md:
```diff
@@ -94,14 +94,15 @@ Things you may want to cover:
+| Column          | Type       | Options                        |
+| -------         | ---------- | ------------------------------ |
+| postal_code     | string     | null: false                    |
+| prefecture_id   | string     | null: false                    |
```
| prefecture_id   | string     | null: false                    |
こちらinteger型にいたしましょう！

yasukuro6259 replied:
修正しました！

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認させていただきました。
追加でコメントいたしましたので、ご確認お願いします！

yasukuro6259 committed:
README修正しました。（3回目）

addressesテーブルの都道府県カラムをinteger型に修正しました。

Katsuno0814 reviewed (commented):
レビュー依頼ありがとうございます！
LGTMとさせていただきます！実装お疲れ様でした！
引き続き実装頑張っていきましょう！
